src/app.ts

```diff
@@ -183,9 +187,47 @@ export async function newApp(
   registerAPIRoutes(app, APIBasePath, allowNotifyIPSourceRange);
   registerPagoPARoutes(app, PagoPABasePath, allowPagoPAIPSourceRange);
 
+  const timer = idpMetadataUpdater(app);
+  app.on("server:stop", () => {
+    clearInterval(timer);
+  });
   return app;
 }
 
+export function idpMetadataUpdater(
+  app: Express,
+  listener?: () => void
+): NodeJS.Timer {
+  return setInterval(async () => {
+    log.info("Restart spid strategy initialization ...");
+
+    container.cache.delete(SPID_STRATEGY);
+
+    const strategy = await container.resolve<Promise<passport.Strategy>>(
+      SPID_STRATEGY
+    );
+    passport.unuse("spid");
+    passport.use("spid", strategy);
+
+    const spidAuth = passport.authenticate("spid", { session: false });
+
+    // Override /login route with new updated spidAuth middleware
+    // tslint:disable: no-any
+    app._router.stack.reduce((prev: ReadonlyArray<any>, _: any) => {
```

Comment:
mmh not sure what you're doing here

`reduce` returns a new array, but it doesn't get assigned to anything

also, I believe this `reduce` can be replaced by a call to `filter`

Reply:
fixed in 82e8752